test/functional/test_framework/test_node.py

```diff
@@ -324,13 +325,13 @@ def is_node_stopped(self):
         return True
 
     def wait_until_stopped(self, timeout=BITCOIND_PROC_WAIT_TIMEOUT):
-        wait_until(self.is_node_stopped, timeout=timeout)
+        wait_until(self.is_node_stopped, timeout=timeout, lock=None, factor=self.factor)
 
     @contextlib.contextmanager
     def assert_debug_log(self, expected_msgs, unexpected_msgs=None, timeout=2):
         if unexpected_msgs is None:
             unexpected_msgs = []
-        time_end = time.time() + timeout
+        time_end = time.time() + int(timeout * self.factor)
```

Comment:
```suggestion
        time_end = time.time() + timeout * self.factor
```

why is this needed?

Reply:
I had problems in test_node.py's *wait_for_rpc_connection*. More precisely, the polling for-loop that refused to iterate over float values. Hence the conversion to int. But if there is a way to make it work without conversion, I'll gladly try it out. Right now, I am going through your list of nits and adapting the code.

Reply:
`time.time()` is a float, so I don't see how this can cause issues